lib/internal/streams/end-of-stream.js

```diff
@@ -53,6 +65,17 @@ function eos(stream, options, callback) {
   const writable = options.writable ||
     (options.writable !== false && isWritableNodeStream(stream));
 
+  if (isNodeStream(stream)) {
+    // Do nothing...
+  } if (isReadableStream(stream) || isWritableStream(stream)) {
+    if (!Duplex) {
+      Duplex = require('./duplex');
+    }
+    stream = Duplex.from(stream);
```

Comment:
Are you sure this does not have any unwanted consequences?

Reply:
In what way?

Reply:
Removed.